src/lib/comms/include/sol-netctl.h

```diff
@@ -76,6 +76,95 @@ typedef struct sol_netctl_network_params {
 } sol_netctl_network_params;
 
 /**
+ * @enum sol_netctl_agent_error_type
+ *
+ * @brief agent error type
+ *
+ * When the network connection failure,
+ * One of these must be chosen to return for connection.
+ * According to the return type, the failure network
+ * connection will be tried again or not.
+ */
+enum sol_netctl_agent_error_type {
+    /*
+     * @brief the error type is retry
+     *
+     * When the network connection failure,
+     * the user can select the return type
+     * and the connection will be tried again.
+     */
+    SOL_NETCTL_AGENT_RETRY,
+    /*
+     * @brief the error type is noretry
+     *
+     * When the network connection failure,
+     * the user can select the return type
+     * and the connection will not be tried again.
+     */
+    SOL_NETCTL_AGENT_NORETRY,
+};
```

Comment:
I'm not convinced we need this as an enum. If it's a simple boolean, go with that. If you think this may grow bigger, then use a struct with a boolean field in it.